### PR TITLE
Don't extend `Base.convert`; instead, define a separate `mbd_unpack` function

### LINKED ISSUE #29: type piracy in `convert` causes ambiguity errors
This definition of `convert` is type piracy, since LMDB doesn't own the `Ref` type:

https://github.com/wildart/LMDB.jl/blob/6212c617976289f8ee0cbeca52a4f3c73114e0b6/src/common.jl#L12

Among the usual problems with type piracy, this also causes ambiguities for `convert(Any, ::Ref{MDB_val})`. Caught in the PkgEval run for https://github.com/JuliaLang/julia/pull/43671, see: https://s3.amazonaws.com/julialang-reports/nanosoldier/pkgeval/by_hash/3656ddf_vs_1db8b8f/LMDB.primary.log

I'd recommend just removing this definition.

### PR DESCRIPTION
**Pull request by DilumAluthge**
Fixes #29

In https://github.com/wildart/LMDB.jl/issues/29#issuecomment-1021493204, @meggart mentioned that this would be a breaking change. Therefore, I have bumped the version number in the `Project.toml` file from 0.2.1 to 1.0.0-DEV to reflect the breaking change.

### CODE REVIEW

File: src/common.jl [hunk 1]
```diff
@@ -9,15 +9,15 @@ function MDBValue(val::T) where {T}
...
 end
 
-convert(::Type{T}, mdb_val_ref::Ref{MDB_val}) where {T} = _convert(T, mdb_val_ref[])
-function _convert(::Type{String}, mdb_val::MDB_val)
+mbd_unpack(::Type{T}, mdb_val_ref::Ref{MDB_val}) where {T} = _mbd_unpack(T, mdb_val_ref[])
+function _mbd_unpack(::Type{T}, mdb_val::MDB_val) where {T <: String}
     unsafe_string(convert(Ptr{UInt8}, mdb_val.mv_data), mdb_val.mv_size)
 end
-function _convert(::Type{Vector{T}}, mdb_val::MDB_val) where {T}
+function _convert(::Type{V}, mdb_val::MDB_val) where {T, V <: Vector{T}}
```
Comment (meggart) [inline src/common.jl]: Is there a reason not to rename this `_convert`?

**Review by meggart — CHANGES_REQUESTED** [review]
Thanks a lot for tackling this issue. Do you still need support in investigating why the tests still fail on 1.8?

**Comment by DilumAluthge** [comment]
I will definitely need help with the test failures.

File: src/common.jl (hunk 1 above)
Comment (DilumAluthge) [reply src/common.jl]: Nope, I just somehow missed it.

**Comment by DilumAluthge** [comment]
Alright, CI is green now.

**Review by meggart — APPROVED** [review]

**Comment by meggart** [comment]
Greate to see the tests passing and again thanks a lot for the fix.

**Pull request merged**